spec/calendar.html

```diff
@@ -287,37 +287,25 @@ <h1>ToWeekOfYear ( _year_, _month_, _day_ )</h1>
     </emu-clause>
 
-    <emu-clause id="sec-temporal-resolvenonlunisolarmonth" aoid="ResolveNonLunisolarMonth">
-      <h1>ResolveNonLunisolarMonth ( _fields_ )</h1>
+    <emu-clause id="sec-temporal-resolvenonlunisolarmonth" aoid="ResolveISOMonth">
+      <h1>ResolveISOMonth ( _fields_ )</h1>
       <p>
-        The abstract operation ResolveNonLunisolarMonth merges the `month` and `monthCode` properties of the given _fields_ Object into an integer month, and validates that they match.
-        It returns a new Object with the same enumerable own String properties as _fields_, adjusted if necessary.
+        The abstract operation ResolveISOMonth merges the `month` and `monthCode` properties of the given _fields_ Object into an integer month, and validates that they match.
+        It returns the integer month.
       </p>
       <emu-alg>
         1. Let _month_ be ? Get(_fields_, *"month"*).
         1. Let _monthCode_ be ? Get(_fields_, *"monthCode"*).
         1. If _monthCode_ is *undefined*, then
           1. If _month_ is *undefined*, throw a *TypeError* exception.
-          1. Set _monthCode_ to ? ToString(_month_).
-        1. Else,
-          1. If Type(_monthCode_) is not String, throw a *RangeError* exception.
-          1. Let _numberPart_ be ! ToNumber(_monthCode_).
-          1. If _numberPart_ is *NaN*, throw a *RangeError* exception.
-          1. If _month_ is not *undefined*, and _month_ ≠ _numberPart_, then
-            1. Throw a *RangeError* exception.
-          1. If ! SameValueNonNumeric(_monthCode_, ! ToString(_numberPart_)) is *false*, then
-            1. Throw a *RangeError* exception.
-          1. Set _month_ to _numberPart_.
-        1. Let _merged_ be ! OrdinaryObjectCreate(%Object.prototype%).
-        1. Let _keys_ be ? EnumerableOwnPropertyNames(_fields_, ~key~).
-        1. For each element _nextKey_ of _keys_, do
-          1. If _nextKey_ is not *"month"* or *"monthCode"*, then
-            1. Let _propValue_ be ? Get(_fields_, _nextKey_).
-            1. If _propValue_ is not *undefined*, then
-              1. Perform ! CreateDataPropertyOrThrow(_merged_, _nextKey_, _propValue_).
-        1. Perform ! CreateDataPropertyOrThrow(_merged_, *"month"*, _month_).
-        1. Perform ! CreateDataPropertyOrThrow(_merged_, *"monthCode"*, _monthCode_).
-        1. Return _merged_.
+          1. Return _month_.
+        1. Let _numberPart_ be ! ToNumber(_monthCode_).
+        1. If _numberPart_ is *NaN*, throw a *RangeError* exception.
+        1. If _month_ is not *undefined*, and _month_ ≠ _numberPart_, then
+          1. Throw a *RangeError* exception.
+        1. If ! SameValueNonNumeric(_monthCode_, ! ToString(_numberPart_)) is *false*, then
```

Comment:
Is this redundant with the previous check? It seems to me that this is unreachable ~code~ abstract.

Reply:
This is needed to make sure that `monthCode` isn't `"     11 "`.